cmd/ocm/gcp/create-wif-config.go

```diff
@@ -0,0 +1,515 @@
+	ctx := context.Background()
+
+	gcpClient, err := gcp.NewGcpClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to initiate GCP client: %v", err)
+	}
+
+	log.Println("Creating workload identity configuration...")
+	wifConfig, err := createWorkloadIdentityConfiguration(models.WifConfigInput{
+		DisplayName: CreateWorkloadIdentityConfigurationOpts.Name,
+		ProjectId:   CreateWorkloadIdentityConfigurationOpts.Project,
+	})
+	if err != nil {
+		log.Fatalf("failed to create WIF config: %v", err)
+	}
+
+	poolSpec := gcp.WorkloadIdentityPoolSpec{
+		PoolName:               wifConfig.Status.WorkloadIdentityPoolData.PoolId,
+		ProjectId:              wifConfig.Status.WorkloadIdentityPoolData.ProjectId,
+		Jwks:                   wifConfig.Status.WorkloadIdentityPoolData.Jwks,
+		IssuerUrl:              wifConfig.Status.WorkloadIdentityPoolData.IssuerUrl,
+		PoolIdentityProviderId: wifConfig.Status.WorkloadIdentityPoolData.IdentityProviderId,
+	}
+	if err = createWorkloadIdentityPool(ctx, gcpClient, poolSpec, CreateWorkloadIdentityConfigurationOpts.DryRun); err != nil {
+		log.Fatalf("Failed to create workload identity pool: %s", err)
+	}
+
+	if err = createWorkloadIdentityProvider(ctx, gcpClient, poolSpec, CreateWorkloadIdentityConfigurationOpts.DryRun); err != nil {
+		log.Fatalf("Failed to create workload identity provider: %s", err)
+	}
+
+	if err = createServiceAccounts(ctx, gcpClient, wifConfig, CreateWorkloadIdentityConfigurationOpts.DryRun); err != nil {
+		log.Fatalf("Failed to create IAM service accounts: %s", err)
+	}
+
+}
+
+func validationForCreateWorkloadIdentityConfigurationCmd(cmd *cobra.Command, argv []string) {
+	if CreateWorkloadIdentityConfigurationOpts.Name == "" {
+		panic("Name is required")
+	}
+	if CreateWorkloadIdentityConfigurationOpts.Project == "" {
+		panic("Project is required")
+	}
+
+	if CreateWorkloadIdentityConfigurationOpts.TargetDir == "" {
+		pwd, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("Failed to get current directory: %s", err)
+		}
+
+		CreateWorkloadIdentityConfigurationOpts.TargetDir = pwd
+	}
+
+	fPath, err := filepath.Abs(CreateWorkloadIdentityConfigurationOpts.TargetDir)
+	if err != nil {
+		log.Fatalf("Failed to resolve full path: %s", err)
+	}
+
+	sResult, err := os.Stat(fPath)
+	if os.IsNotExist(err) {
+		log.Fatalf("Directory %s does not exist", fPath)
+	}
+	if !sResult.IsDir() {
+		log.Fatalf("file %s exists and is not a directory", fPath)
+	}
+
+}
+
+func createWorkloadIdentityConfiguration(input models.WifConfigInput) (*models.WifConfigOutput, error) {
+	// TODO: Implement the actual creation of the workload identity configuration
+	return mockWifConfig(), nil
+}
+
+func createWorkloadIdentityPool(ctx context.Context, client gcp.GcpClient, spec gcp.WorkloadIdentityPoolSpec, generateOnly bool) error {
+	name := spec.PoolName
+	project := spec.ProjectId
+	if generateOnly {
+		log.Printf("Would have created workload identity pool %s", name)
+	} else {
```

Comment:
```suggestion
	if generateOnly {
		log.Printf("Would have created workload identity pool %s", name)
		return nil
	} 
```
Doing it this way removes the need for an else branch. WDYT?